system/pam_limits.py

```diff
@@ -191,11 +197,23 @@ def main():
                 nf.write(line)
                 continue
 
+            actual_value_unlimited = actual_value in ['unlimited', 'infinity', '-1']
+
             if use_max:
-                new_value = max(value, actual_value)
+                if value.isdigit() and actual_value.isdigit():
+                    new_value = max(value, actual_value)
+                elif actual_value_unlimited:
+                    new_value = actual_value
+                else:
+                    new_value = value
 
             if use_min:
-                new_value = min(value,actual_value)
+                if value.isdigit() and actual_value.isdigit():
+                    new_value = min(value, actual_value)
+                elif actual_value_unlimited:
+                    new_value = value
+                else:
+                    new_value = actual_value
 
```

Comment:
In order to keep idempotence in case both actual_value and value are set to different unlimited strings, I think the min check needs to check whether the value is an unlimited version and set to actual_value if so, otherwise fall back to new_value = value.  (that way we always prefer the current configuration if they are different strings but equivalent).